### PR TITLE
bugfix: should use shard' domain notification version

### PR DESCRIPTION
**Pull request by wxing1292**

### CODE REVIEW

File: service/history/timerQueueProcessor.go [hunk 1]
```diff
@@ -135,9 +135,9 @@ func (t *timerQueueProcessorImpl) FailoverDomain(domainID string) {
 			standbyClusterName = cluster
 		}
 	}
-
 	// the ack manager is exclusive, so just add a cassandra min presicition
 	maxLevel := t.activeTimerProcessor.timerQueueAckMgr.getReadLevel().VisibilityTimestamp.Add(1 * time.Millisecond)
+	t.logger.Infof("Timer Failover Triggered: %v, min level: %v, max level: %v.\n", domainID, minLevel, maxLevel)
```
Comment (samarabbas) [inline service/history/timerQueueProcessor.go]: Can we have domainID as a tag?

File: service/history/transferQueueProcessor.go [hunk 2]
```diff
@@ -147,8 +147,10 @@ func (t *transferQueueProcessorImpl) FailoverDomain(domainID string) {
 			standbyClusterName = cluster
 		}
 	}
+
 	// the ack manager is exclusive, so add 1
 	maxLevel := t.activeTaskProcessor.getReadLevel() + 1
+	t.logger.Infof("Transfer Failover Triggered: %v, min level: %v, max level: %v.\n", domainID, minLevel, maxLevel)
```
Comment (samarabbas) [inline service/history/transferQueueProcessor.go]: Let's have domainID as a tag.

File: service/history/historyEngine.go [hunk 3]
```diff
@@ -199,9 +199,16 @@ func (e *historyEngineImpl) registerDomainFailoverCallback() {
 		e.shard.GetShardID(),
 		e.shard.GetDomainCache().GetDomainNotificationVersion(),
 		func(prevDomain *cache.DomainCacheEntry, nextDomain *cache.DomainCacheEntry) {
+			domainFailoverNotificationVersion := nextDomain.GetFailoverNotificationVersion()
+			shardNotificationVersion := e.shard.GetDomainNotificationVersion()
+			domainActiveCluster := nextDomain.GetReplicationConfig().ActiveClusterName
+
+			e.logger.Infof("Domain Failover Event: Shard: %v, Domain: %v, ID: %v, Failover Notification Version: %v, Active Cluster: %v, Shard Domain Notification Version: %v\n",
```
Comment (samarabbas) [inline service/history/historyEngine.go]: Can you convert these parameters into tags?

File: service/history/timerQueueProcessor.go (hunk 1 above)
Comment (wxing1292) [reply service/history/timerQueueProcessor.go]: sure

File: service/history/historyEngine.go (hunk 3 above)
Comment (wxing1292) [reply service/history/historyEngine.go]: this log will only appears once, per shard.
the domain id tagged logger is in each processor

**Pull request merged**